File: SQLiteDBTuto/app/src/main/java/com/example/sqlitedbtuto/MainActivity.java
package com.example.sqlitedbtuto;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Display;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ListView myListView;
    private Button addBtn;
    DBHelper mydb;
    ArrayAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mydb = new DBHelper(this);
        ArrayList arrayList = mydb.getAllMovies();
        mAdapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, arrayList);

        myListView = findViewById(R.id.listView1);
        myListView.setAdapter(mAdapter);
        addBtn = findViewById(R.id.btnAdd);

        //리스트 뷰 item update를 위한 리스너
        myListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long arg4) {
                String item = (String) parent.getItemAtPosition(position);
                Log.i("setOnItemClickListener", "parent.getItemAtPosition(position) : " + item);
                String[] strArray = item.split(" ");
                int id = Integer.parseInt(strArray[0]);
                Bundle dataBundle = new Bundle();
                dataBundle.putInt("id", id);
                Intent intent = new Intent(getApplicationContext(), DisplayMovie.class);
                intent.putExtras(dataBundle);
                startActivity(intent);
                Log.i("main 추가 버튼", "끝");
            }
        });

        //영화 추가 및 삭제
        addBtn.setOnClickListener(new Button.OnClickListener() {
            @Override
            public void onClick(View view) {
                Bundle bundle = new Bundle();
                bundle.putInt("id", 0);
                Intent intent = new Intent(getApplicationContext(), DisplayMovie.class);
                intent.putExtras(bundle);
                startActivity(intent);
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        mAdapter.clear();
        mAdapter.addAll(mydb.getAllMovies());
        mAdapter.notifyDataSetChanged();
    }
}

File: DataBindingTuto/app/src/main/java/com/example/databindingtuto/MainActivity.kt
package com.example.databindingtuto

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.databinding.DataBindingUtil
import com.example.databindingtuto.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    var hello = "Hello World!"
    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)// 잊지마라 activity에서 setContentview를 id로하면 layout이 이중으로 inflate 됨
//        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.activity = this
        hello = "create"
    }

    override fun onPause() {
        super.onPause()
        hello += "pause"
    }
    override fun onStart() {
        super.onStart()
        hello += "start"
    }
    fun tvClick(view: View) {
        Log.i("tvClick", "hello: $hello")
        hello += "click"
        if (hello.length >= 15)
            hello = "clear"
        binding.activity = this
    }
}